web/src/markdown.js

```diff
@@ -16,11 +16,10 @@ import marked from "../third/marked/lib/marked";
 // If we see preview-related syntax in our content, we will need the
 // backend to render it.
 const preview_regexes = [
-    // Inline image previews, check for contiguous chars ending in image suffix
+    // Inline image and video previews, check for contiguous chars ending in image and video suffix
     // To keep the below regexes simple, split them out for the end-of-message case
 
-    /\S*(?:\.bmp|\.gif|\.jpg|\.jpeg|\.png|\.webp)\)?\s+/m,
-    /\S*(?:\.bmp|\.gif|\.jpg|\.jpeg|\.png|\.webp)\)?$/m,
+    /\S*(?:\.bmp|\.gif|\.jpg|\.jpeg|\.png|\.webp|\.mp4|\.webm)\)?(\s+|$)/m,
```

Comment:
How was the set of video file types to support selected?

Reply:
I believe that the idea here is to support only video formats (containers) that are supported cross-browser and cross-device; Mozilla maintains a list of those in a table that can be found at https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Containers#index_of_media_container_formats_file_types -- as you can see there, MP4 and WebM are the only formats that fit the bill in terms of interoperability.

Reply:
@amanagr can you paste this explanation and reference URL into a comment (probably not here, but in the backend implementation)? Seems worth having that written down for posterity.

Reply:
Done

Reply:
It would be nice to allow the admin to provide this list, regardless of what's commonly supported.

Reply:
Probably some change in this space is a good idea but let's leave that discussion for a follow-up project once this merges; I'm not sure what the best way to do such a setting is and I think it's best to make sure we can actually integrate the base feature here.